app/staking/components/stakingModal/StakingModal.tsx

```diff
@@ -0,0 +1,299 @@
+"use client";
+import Container from "@/components/container/container";
+import Spacer from "@/components/layout/spacer";
+import Modal from "@/components/modal/modal";
+import {
+  UserUnbondingDelegation,
+  Validator,
+  ValidatorWithDelegations,
+} from "@/hooks/staking/interfaces/validators";
+import { GetWalletClientResult } from "wagmi/actions";
+import styles from "./StakingModal.module.scss";
+import Text from "@/components/text";
+import Icon from "@/components/icon/icon";
+import { formatBalance } from "@/utils/formatting/balances.utils";
+import Input from "@/components/input/input";
+import Button from "@/components/button/button";
+import { ChangeEvent, SetStateAction, useState } from "react";
+import Tabs from "@/components/tabs/tabs";
+import { StakingTxTypes } from "@/hooks/staking/interfaces/stakingTxTypes";
+import { StakingTabs } from "../stakingTab/StakingTabs";
+import { getBalanceForValidator } from "@/hooks/staking/helpers/userStaking";
+import Selector, { Item } from "@/components/selector/selector";
+import Amount from "@/components/amount/amount";
+
+export interface StakingModalParams {
+  validator: Validator | null;
+  userStaking?: {
+    validators: ValidatorWithDelegations[];
+    unbonding: UserUnbondingDelegation[];
+    cantoBalance: string;
+  };
+  signer: GetWalletClientResult | undefined;
+  onConfirm: (
+    validator: Validator | null,
+    inputAmount: string,
+    selectedTx: StakingTxTypes,
+    validatorToRedelegate: Validator | null | undefined
+  ) => void;
+  validators: Validator[];
+}
+export const StakingModal = (props: StakingModalParams) => {
+  const [inputAmount, setInputAmount] = useState("");
+  const [maxClicked, setMaxClicked] = useState(false);
+
+  const [selectedTx, setSelectedTx] = useState<StakingTxTypes>(
+    StakingTxTypes.DELEGATE
+  );
+  const [activeTab, setActiveTab] = useState("delegate");
+  const [validatorToRedelegate, setValidatorToRedelegate] =
+    useState<Validator | null>();
+
+  const splicedValidators = props.validators.filter(
+    (validator) =>
+      validator.operator_address !== props.validator?.operator_address
+  );
+
+  const dropdownItems = splicedValidators.map((validator) => {
```

Comment:
just combine this with splicedValidators. No need for two variables to define the same thing.